File: core/src/io/cucumber/utils/helper/NumbersHelper.java
package io.cucumber.utils.helper;

import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.utils.Array;

import java.util.Arrays;
import java.util.List;

public class NumbersHelper {

    private static List<Texture> numbersSymbols = Arrays.asList(
            new Texture("0.png"),
            new Texture("1.png"),
            new Texture("2.png"),
            new Texture("3.png"),
            new Texture("4.png"),
            new Texture("5.png"),
            new Texture("6.png"),
            new Texture("7.png"),
            new Texture("8.png"),
            new Texture("9.png")
    );

    public static Array<Texture> getTextures(int value) {
        Array<Texture> textures = new Array<Texture>();
        if (value == 0) {
            textures.add(numbersSymbols.get(0));
        }

        while (value > 0) {
            int number = value % 10;
            textures.add(numbersSymbols.get(number));
            value /= 10;
        }
        textures.reverse();

        return textures;
    }

    private NumbersHelper() {
    }
}

File: core/src/io/cucumber/utils/NumbersHelper.kt
package io.cucumber.utils

import com.badlogic.gdx.graphics.Texture

object NumbersHelper {

    private val numbersSymbols = mapOf(
        '1' to Texture("1.png"),
        '2' to Texture("2.png"),
        '3' to Texture("3.png"),
        '4' to Texture("4.png"),
        '5' to Texture("5.png"),
        '6' to Texture("6.png"),
        '7' to Texture("7.png"),
        '8' to Texture("8.png"),
        '9' to Texture("9.png"),
        '0' to Texture("0.png")
    )

    fun getTextures(value: Int): List<Texture> =
        value.toString().toCharArray().map { numbersSymbols[it]!! }

}